ddtrace/tracer/civisibility_payload.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"bytes"
+	"sync/atomic"
+
+	"github.com/tinylib/msgp/msgp"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+)
+
+// civisibilitypayload represents a payload specifically designed for CI Visibility events.
+// It embeds the generic payload structure and adds methods to handle CI Visibility specific data.
+type civisibilitypayload struct {
```

Comment:
Nit: we should use camel-case for the type name.

```suggestion
type ciVisibilityPayload struct {
```

Reply:
fixed in [465456a](https://github.com/DataDog/dd-trace-go/pull/2739/commits/465456aeed42aa1b9a6b4a96c0dad5a402e09308)